kore/src/Kore/Log/ErrorRewriteLoop.hs

```diff
@@ -27,23 +38,32 @@ import Kore.Syntax.Variable
 
 import Log
 
-newtype ErrorRewriteLoop =
-    ErrorRewriteLoop { getRule :: RewriteRule Variable }
+data ErrorRewriteLoop =
+    ErrorRewriteLoop
+        { rule :: RewriteRule Variable
+        , errorCallStack :: CallStack
+        }
+    deriving (Show)
+
+instance Exception ErrorRewriteLoop
```

Comment:
We can use a custom `Exception` instance to ensure that the entry is always wrapped:

```suggestion
instance Exception ErrorRewriteLoop where
    toException = toException . SomeEntry
    fromException exn = fromException exn >>= \(SomeEntry entry) -> pure entry
```

We will also have to add `instance Exception SomeEntry` where `SomeEntry` is defined.